torchdynamo/convert_frame.py

```diff
@@ -197,24 +214,41 @@ def format_error_msg(exc, frame):
                 + "=" * 10
                 + "\n\n"
             )
+            stack_above_dynamo = []
+            if frame is not None:
+                stack_above_dynamo = filter_stack(traceback.extract_stack(frame))
+
             msg += "".join(
                 traceback.format_list(
-                    filter_stack(traceback.extract_stack(frame))
-                    + list(reversed(exc.real_stack))
+                    stack_above_dynamo + list(reversed(exc.real_stack))
                 )
             )
+
+        msg += replay_record_msg()
+
     else:
-        msg = f"WON'T CONVERT {frame.f_code.co_name} {frame.f_code.co_filename}\
- line {frame.f_code.co_firstlineno} \ndue to: \n{traceback.format_exc(limit=-1)}"
+        msg = f"WON'T CONVERT {code.co_name} {code.co_filename}\
+ line {code.co_firstlineno} \ndue to: \n{traceback.format_exc(limit=-1)}"
 
         if hasattr(exc, "real_stack"):
             msg += f"\nfrom user code:\n {''.join(traceback.format_list([exc.real_stack[-1]]))}"
 
+        msg += replay_record_msg()
+
         msg += "\nSet torchdynamo.config.verbose=True for more information\n"
     msg += "=" * 10
     return msg
 
 
+def exception_handler(e, code, frame=None):
```

Comment:
Not super important, but writing files in the middle of an exception handling stack feels weird. Would we lose anything if this were called from finally instead?

Reply:
I can do this, 

> Can we add tests for this? A good bit of logic here in core code paths that I don't think would be easy to ensure we do not rot away without tests.

Yeah I'm definitely going to add tests for this, using eval or something.

Reply:
I take this back, a finally would run whether an error gets thrown or not, I don't think there is another place that this can be handled from. If you're worried about writing a file throwing another exception, I can wrap this in nested try/catch and print a warning if there is an issue with writing the file.

Reply:
Handled the file operations with a wrapping try/except